### PR TITLE
Update docs with hs owned engagements event handlers

### PR DESCRIPTION
**Pull request by esme**
## Description

Jira Ticket: CALL-5011

## Merge Checklist

| Q                        | A 
| ------------------------ | --------------
| Adds Documentation?      | yes
| Any Dependency Changes?  |
| Patch: Bug Fix?          |
| Minor: New Feature?      |
| Major: Breaking Change?  |

[BRAVE Checklist](https://github.com/HubSpot/calling-extensions-sdk/blob/master/SHIP_WITH_CARE.md)

- [x] I have read the BRAVE checklist and confirmed if the following is necessary.

| Q                              | A 
| ------------------------------ | --------------
| Backwards Compatible?          |
| Rollout/Rollback Plan?         | 
| Automated test coverage?       | 
| Verified that changes work?    |
| Expect Dependencies to Fail?   |

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,9 +122,20 @@ const options = {
...
       /* HubSpot has created an engagement for this call. */
+    }
+    onEngagementCreatedFailed: event => {
+      /* HubSpot has failed to create an engagement for this call. */
+    }
+    onUpdateEngagementSucceeded: event => {
+      /* HubSpot has updated an engagement for this call. */
     },
+    onUpdateEngagementFailed: event => {
+      /* HubSpot has failed to updat an engagement for this call. */
```
Comment (jmp3833) [inline README.md:137]: Nit: `updat` -> `update`

File: README.md [hunk 2]
```diff
@@ -329,20 +341,68 @@ onDialNumber(data) {
...
+
+<details>
+ <summary>onCreateEngagementFailed</summary>
+ <p>
+
+```js
+  // Message indicating that HubSpot has failed to create an engagement
+  onCreateEngagementFailed(data) {
+    const {
+      error: Error
```
Comment (jmp3833) [inline README.md:368]: Is there any public documentation around the /engagements API we could reference around what fields the Error object may contain?

**Review by jmp3833 — APPROVED** [review]
Thanks @esme

File: README.md (hunk 2 above)
Comment (esme) [reply README.md:368]: We're using the built-in Javascript [Error() constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) currently but I think we might create an error wrapper to standardize this in the future. What do you think of using an error object for the time being?
```
const {
  error: { message: string }
}
```

Comment (jmp3833) [reply README.md]: That SGTM @esme, we can get more specific with error types over time

**Review by jmp3833 — APPROVED** [review]
LGTM @esme - thanks for bumping the SDK version here as well

**Pull request merged**